### DISCUSSION TITLE
Trimbox: Internationalization and Localization for Extension Pay

### DISCUSSION THREAD

**Discussion by jordangaston** [Glench/ExtPay#29] [General]
The goal of this ask is for ExtensionPay to support internationalization and localization of...

### The Payment Page
Ideally we would be able to control...
- internationalization of the extension name and form elements
- internationalization of the currency
- localization of the price

![Screen Shot 2022-06-08 at 08 24 58](https://user-images.githubusercontent.com/14916087/172627657-63aadfb7-cbc2-42fc-ae41-e3524dc73164.png)

### The Log In Page
![Screen Shot 2022-06-08 at 08 29 31](https://user-images.githubusercontent.com/14916087/172630782-1e816470-34ab-410d-a369-e3ebd6caccfa.png)

### The Login Email
![Screen Shot 2022-06-08 at 08 32 17](https://user-images.githubusercontent.com/14916087/172629372-af5166b6-329c-4380-8d82-20ebd2efc68d.png)

### The Success and Error Messages
![Screen Shot 2022-06-08 at 08 33 22](https://user-images.githubusercontent.com/14916087/172630225-94d06e0f-310f-4096-ae38-1b9392c7bd12.png)
![Screen Shot 2022-06-08 at 08 34 29](https://user-images.githubusercontent.com/14916087/172630327-4c16b044-6987-451d-9e8a-8d3dac515575.png)

### The Reciept

![Screen Shot 2022-06-08 at 08 41 10](https://user-images.githubusercontent.com/14916087/172631410-b056c724-dc42-4c99-ab61-6913469b39cd.png)

**Comment by jordangaston** [comment]
Hey @Glench I just saw that you moved this issue into a discussion.  Any initial thoughts on this ask?

**Comment by henrikra** [comment]
Localization indeed should be done.

Users get confused now because my extension is not in English but the payment page is. This lowers the trust from users and leads them to not subscribe for the service.

@Glench Have you looked into this?

**Comment by Glench** [comment]
So there's a lot here. To me there are three separate issues:

1. Pricing / currency options
2. Translation (All the text that a user sees. To me this is the easiest one and is already mostly done.)
3. Extension name (because I can't automatically translate your extension name)

I'm especially interested in the first but there are some questions about the second and third. Got any proposals for how it should work?

**Reply by henrikra** [reply]
Yeah what I meant with localization is the 2nd on your list 👍 

It is very important that the extension is in the same language as the payment page 👍 

How much is left as you said it is mostly done? Do we pass some language param to the paymentPage function?

**Comment by Glench** [comment]
@jordangaston how are you detecting the language in your extension?

And for pricing — I've gotten requests from others for charging only in a specific currency but it sounds like you may want to somehow convert your extension's pricing to a user's local currency? Any thoughts about how you'd like to do that?

**Reply by henrikra** [reply]
The language should be passed by extension to ExtPay as a string when calling for example `openPaymentPage`.

**Reply by Glench** [reply]
Yeah, that seems like one good option, but what about the case where it's not passed? I'd like to know how to translate the page in that case.

**Reply by henrikra** [reply]
ExtPay should use English as default if language is not passed. So the language param is optional 👍

**Reply by burkybang** [reply]
Or it could get the user's language preference from `navigator.language` or `navigator.languages` as the default

**Reply by henrikra** [reply]
Yeah that could be the default but then I can override that value by passing language like this `openPaymentPage({language: 'es'})`

**Reply by Glench** [reply]
@henrikra This is a good idea. I'm currently looking for translators for ExtensionPay's screens. Are you able to translate anything?

**Reply by henrikra** [reply]
@Glench Sorry I stopped using this library for now. Implemented my own system 👍

**Reply by surtic86** [reply]
> @henrikra This is a good idea. I'm currently looking for translators for ExtensionPay's screens. Are you able to translate anything?

i could Translate in German. Are you using any Translation Tool? For myself i use Weblate so Users could easy made changes to it.

**Reply by davioliveiraweb** [reply]
Any update about this?

**Comment by surtic86** [comment]
Yes would also like to see this.

**Comment by Glench** [comment]
I guess I never updated this thread, but the payment page has been translated into several languages for oh maybe a year or so now. More languages and pages should be coming, too.

**Reply by davioliveiraweb** [reply]
Was it translated to Brazilian Portuguese (pt_BR)?

**Reply by Glench** [reply]
Yes
